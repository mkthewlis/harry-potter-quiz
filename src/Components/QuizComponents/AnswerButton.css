.answer-button {
    min-height: 1.8rem;
    min-width: 40rem;
    background-color: #FFFFFF;
    border-radius: 10px;
    align-items: center;
    cursor: pointer;
    border: 2px solid #FFC500;
    color: #7F0909;
    font-size: 1.5em;
    padding: .5rem;
    margin: 1rem 0;
    font-family: 'Courier New', Courier, monospace;
}