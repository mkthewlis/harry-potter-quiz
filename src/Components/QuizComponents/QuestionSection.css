.question-section__statistics-container {
    display: flex;
    justify-content: center;
    align-items: center;
}

.question-section__statistics-box {
    min-width: 5rem;
    min-height: 1.5rem;
    background-color: #FFC500;
    padding: 0.5rem;
    color: #7F0909;
    border: 2px solid #591313;
    border-radius: 5px;
    font-size: 1.3em;
    justify-content: center;
    margin-right: 5px;
}

.question-section__question-number {
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 3em;
}

.question-section__question {
    font-size: 2em;
}

.question-section__answer-container {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-self: center;
}

.question-section__answer-container button {
    display: flex;
    align-self: center;
    justify-content: center;
}