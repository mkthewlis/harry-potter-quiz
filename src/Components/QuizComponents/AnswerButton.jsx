import React from 'react';
import './AnswerButton.css';

export const AnswerButton = ({index, answer, ...onClick}) => (
    <>
        <button key={index} className="answer-button" {...onClick}>{answer}</button>
    </>
);