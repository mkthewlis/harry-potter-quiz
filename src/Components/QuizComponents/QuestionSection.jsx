import React from 'react';
import { AnswerButton } from './AnswerButton';
import './QuestionSection.css';

export const QuestionSection = ({questionNumber, randomQuestions, countdownTimer, result, handleAnswerButtonClick}) => (
    <section>
        <h2 className="question-section__question-number">Question {questionNumber + 1}/{randomQuestions.length}</h2>
        <div className="question-section__statistics-container">
            <div className="question-section__statistics-box">Your score: {result}</div>
            <div className="question-section__statistics-box">{countdownTimer}s</div>
        </div>
        <p className="question-section__question">{randomQuestions[questionNumber].question}</p>
        <div className="question-section__answer-container">
            {randomQuestions[questionNumber].possibleAnswers.map((possibleAnswer) => (
                <AnswerButton onClick={() => handleAnswerButtonClick(possibleAnswer.correctAnswer)} answer={possibleAnswer.answer}/>
            ))}
        </div>
    </section>
);