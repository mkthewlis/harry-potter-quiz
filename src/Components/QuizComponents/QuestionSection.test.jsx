import React from 'react';
import { mount } from 'enzyme';
import { QuestionSection } from './QuestionSection';
import { AnswerButton } from './AnswerButton';

describe('QuestionSection', () => {
    let wrapper;
    
    const randomQuestionsMock = [
        {
          question: 'What do the initials J.K. Rowling stand for?',
          possibleAnswers: [
            { answer: 'Joanne Kathleen', correctAnswer: true },
            { answer: 'Jane Kathleen', correctAnswer: false },
            { answer: 'Jody Katie', correctAnswer: false },
            { answer: 'Joanne Katie', correctAnswer: false }
          ],
        }
    ];
    const handleAnswerButtonClickMock = jest.fn();
    const questionNumberMock = 0;
    const countdownTimerMock = 15;
    const resultMock = 0;

    beforeEach(() => {
        wrapper = mount(
          <QuestionSection 
            questionNumber={questionNumberMock}
            randomQuestions={randomQuestionsMock}
            countdownTimer={countdownTimerMock}
            result={resultMock} 
            handleAnswerButtonClick={handleAnswerButtonClickMock}/>,
        );
      });
    it('renders without crashing', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the current question number in the title', () => {
        expect(wrapper.find('.question-section__question-number').text()).toContain('Question 1/1');
    });

    it('should display the current result', () => {
        expect(wrapper.find('.question-section__statistics-box').at(0).text()).toContain('Your score: 0');
    });

    it('should display the time left in the coundown timer', () => {
        expect(wrapper.find('.question-section__statistics-box').at(1).text()).toContain('15s');
    });

    it('should display the current question', () => {
        expect(wrapper.find('.question-section__question').text()).toContain('What do the initials J.K. Rowling stand for?');
    });

    it('should display four answer buttons, one for each answer', () => {
        expect(wrapper.find(AnswerButton)).toHaveLength(4);
    });

    it('should not call the handleAnswerButtonClick function by default', () => {
        expect(handleAnswerButtonClickMock).not.toHaveBeenCalled();
    });
});  