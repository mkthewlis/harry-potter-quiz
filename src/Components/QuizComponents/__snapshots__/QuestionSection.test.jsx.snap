// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionSection renders without crashing 1`] = `
<QuestionSection
  countdownTimer={15}
  handleAnswerButtonClick={[MockFunction]}
  questionNumber={0}
  randomQuestions={
    Array [
      Object {
        "possibleAnswers": Array [
          Object {
            "answer": "Joanne Kathleen",
            "correctAnswer": true,
          },
          Object {
            "answer": "Jane Kathleen",
            "correctAnswer": false,
          },
          Object {
            "answer": "Jody Katie",
            "correctAnswer": false,
          },
          Object {
            "answer": "Joanne Katie",
            "correctAnswer": false,
          },
        ],
        "question": "What do the initials J.K. Rowling stand for?",
      },
    ]
  }
  result={0}
>
  <section>
    <h2
      className="question-section__question-number"
    >
      Question 
      1
      /
      1
    </h2>
    <div
      className="question-section__statistics-container"
    >
      <div
        className="question-section__statistics-box"
      >
        Your score: 
        0
      </div>
      <div
        className="question-section__statistics-box"
      >
        15
        s
      </div>
    </div>
    <p
      className="question-section__question"
    >
      What do the initials J.K. Rowling stand for?
    </p>
    <div
      className="question-section__answer-container"
    >
      <AnswerButton
        answer="Joanne Kathleen"
        key="0"
        onClick={[Function]}
      >
        <button
          className="answer-button"
          onClick={[Function]}
        >
          Joanne Kathleen
        </button>
      </AnswerButton>
      <AnswerButton
        answer="Jane Kathleen"
        key="1"
        onClick={[Function]}
      >
        <button
          className="answer-button"
          onClick={[Function]}
        >
          Jane Kathleen
        </button>
      </AnswerButton>
      <AnswerButton
        answer="Jody Katie"
        key="2"
        onClick={[Function]}
      >
        <button
          className="answer-button"
          onClick={[Function]}
        >
          Jody Katie
        </button>
      </AnswerButton>
      <AnswerButton
        answer="Joanne Katie"
        key="3"
        onClick={[Function]}
      >
        <button
          className="answer-button"
          onClick={[Function]}
        >
          Joanne Katie
        </button>
      </AnswerButton>
    </div>
  </section>
</QuestionSection>
`;
